### PR TITLE
t/26 - add Production Pack demo + CKE5/CKBox version bump

### PR DESCRIPTION
**Pull request by usqr**
Closes: #26

### CODE REVIEW

**Review by mlewand — CHANGES_REQUESTED** [review]
The new PP demo needs some tweaks. Document editor UI sample got broken.



When running PP demo, there's a warning:

```
ckeditorerror.js:119 toolbarview-item-unavailable {item: 'formatPainter'} 
```



Changes in CSS, HTML look good (these related to mentions colors, http to https change for image srcs and CKBox link).

Good call with removing ckeditor5 and collaboration-features package resolutions - they were not needed, don't know why they ended up there.

**Comment by mlewand** [comment]
Quite a big PR, it would make sense for you to pass more context what the change is all about.

Since the changes were significant let's also ask our great @ckeditor/qa-team team to test the demos before going with next review round. @usqr can you ask them and handle the communication (including information what was changed and what was the reason for it)?

File: headless/demo.css [hunk 1]
```diff
@@ -60,6 +60,7 @@ html {
   background: white;
   border: 1px solid var(--ck-color-base-border);
   border-radius: var(--ck-demo-space-xs);
+  z-index: 1;
```
[the remark this answers is not in the document]
Comment (usqr) [reply headless/demo.css:63]: due to: cksource/nova-ckeditor.com#1535

**Comment by usqr** [comment]
@mlewand So some background - There are lots of changes going on in nova on demos (bug fixes, feature updates, new demos). With that amount of work and a small team we don't have time to keep public repo up to date at all times so when we have some more time in a sprint we try to bundle those changes.

*   As the public repo should be "near mirror image" of the private one there will be some css/html/7/or content changes changes as we update/optimize/fix demos in the private one - the css/content are copied directly so any changes have to be reflected in the public one
*   there are usual version bumps here
*   new PP demo (kinda self explanatory)
*   new features added (the demos on nova are evolving fast and it is hard to keep up to date with all the changes - especially that this repo is of least importance \[tbh for ours and your teams :shrug: \])

I will squash all the bugs and pass it on to the QA team. :+1:

**Review by pszczesniak — COMMENTED** [review]
General review comment:

*   It's a big PR so lest ask **QA Team** to test the demos,
*   One thing i've noticed - in `Full featured` demo there isn't `Show blocks` feature enabled. (on ckeditor.com also),
*   new **Productivity Pack** demo - for me it looks a bit odd without any information why it does not work out of the box (i would add some info text in `index.html`),
*   similar situation in `user-interface-document` demo, but there is also o visibility issue, when column on the left is empty (it's reserved place for `document outline`),
*   in `Full featured` demo in content we mention about `CKBox`, but (same as above) to make it work we need to look in the code and read about missing license keys - info in `index.html` would be best from my point of view - but of course - discussion needed :slightly\_smiling\_face: 

I've fix sticky toolbar in `Headless` demo.

File: feature-rich/package.json [hunk 2]
```diff
@@ -1,141 +1,143 @@
...
+		"webpack": "^5.75.0",
+		"webpack-cli": "^4.10.0",
+		"webpack-dev-server": "^4.11.1"
+	},
+	"scripts": {
+		"build": "webpack --mode production",
+		"build-dev": "webpack --mode development",
+		"start": "webpack serve --open /index.html --mode development"
+	},
+	"resolutions": {
```
Comment (Reinmar) [inline feature-rich/package.json:80]: https://github.com/ckeditor/ckeditor5-demos/issues/29#issuecomment-1721083665 Resolutins are not necessary any more.

**Comment by usqr** [comment]
> Resolutins are not necessary any more.

Resolutions removed

> *   It's a big PR so lest ask **QA Team** to test the demos,

QA task: cksource/qa-internal#239

> *   One thing i've noticed - in `Full featured` demo there isn't `Show blocks` feature enabled. (on ckeditor.com also),

This Repo just reflects CKEditor.com demos - most of the changes are requested/approved by marketing... :man\_shrugging: 

> *   new **Productivity Pack** demo - for me it looks a bit odd without any information why it does not work out of the box (i would add some info text in `index.html`),
> *   similar situation in `user-interface-document` demo, but there is also o visibility issue, when column on the left is empty (it's reserved place for `document outline`),
> *   in `Full featured` demo in content we mention about `CKBox`, but (same as above) to make it work we need to look in the code and read about missing license keys - info in `index.html` would be best from my point of view - but of course - discussion needed 🙂

This an issue of the fact that we are using premium features in those demos we can't provide licence keys/token/other access data in public facing demo yet we want to have feature parity with the demos on cke.com.  
it's the same situation with source code for the demos in the docs: [https://ckeditor.com/docs/ckeditor5/latest/examples/builds-custom/collaborative-document-editor.html#editor-example-configuration](https://ckeditor.com/docs/ckeditor5/latest/examples/builds-custom/collaborative-document-editor.html#editor-example-configuration)  
But I agree that adding some sort of info in the index files would be good. WDYT @Witoso :thinking:

**Comment by charlttsie** [comment]
@usqr Slash command plugin is not added to the `user-interface-button-grouping` demo code (it's present on ckeditor.com).

**Pull request merged**